src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressRing/ProgressRing.cs

```diff
@@ -244,8 +244,8 @@ private void SetAnimatedVisualPlayerSource(bool force = false)
 					}
 					else
 					{
-						_currentSourceUri = indeterminateSource == null ? FeatureConfiguration.ProgressRing.ProgressRingAsset : indeterminateSource.UriSource;
-						_player.Source.UriSource = _currentSourceUri;
+						_currentSourceUri = (indeterminateSource == null && indeterminateSource is not IAnimatedVisualSourceWithUri) ? FeatureConfiguration.ProgressRing.ProgressRingAsset : (indeterminateSource as IAnimatedVisualSourceWithUri)!.UriSource;
```

Comment:
Probably this?

```suggestion
						_currentSourceUri = indeterminateSource is IAnimatedVisualSourceWithUri indeterminateSourceWithUri ? indeterminateSourceWithUri.UriSource : FeatureConfiguration.ProgressRing.ProgressRingAsset;
```

(This comment applies to code in line 280 as well)

Reply:
look better ! thanks !